src/serializers/htmlToSlate/index.ts

```diff
@@ -81,17 +85,30 @@ const deserialize = ({
   }
 
   if (config.textTags[nodeName] || el.type === ElementType.Text) {
-    const attrs = gatherTextMarkAttributes({ el: parent })
+    const attrs = gatherTextMarkAttributes({ el: currentEl })
     const text = processTextValue({
       text: textContent(el as Element),
       context: childrenContext as Context,
       isInlineStart: index === 0,
       isInlineEnd: Number.isInteger(childrenLength) && isLastChild,
       isNextSiblingBlock: (el.next && isTag(el.next) && isBlock(el.next.tagName)) || false,
     })
-    if ((config.filterWhitespaceNodes && isAllWhitespace(text) && !childrenContext) || text === '') {
+    if (text === '') {
       return null
     }
+    if (isAllWhitespace(text)) {
+      if (config.filterWhitespaceNodes && !childrenContext) {
+        return null
+      }
+      if (config.convertBrToLineBreak) {
+        if (currentEl.prev && getName(currentEl.prev as Element) === 'br') {
+          return null
+        }
+        if (currentEl.next && getName(currentEl.next as Element) === 'br') {
+          return null
+        }
+      }
```

Comment:
If I understand correctly, whitespace characters (`\t`, `\n` and '/r`) that immediately follow or precede a `<br>` will not be processed to Slate in the case where we convert `br` tags to new lines.

The idea is that if both `br` tags and new line characters are used in a text block, the best interpretation (when `br` tag conversion to new line is active) is that the new line character is only used for visual presentation of HTML source.

Reply:
Yeah, that solved the issue with excessive whitespace.
Sorry for not documenting it with a comment in source code